pkg/api/v1/schema.go

```diff
@@ -32,6 +32,20 @@ import (
  ATTENTION: Rerun code generators when comments on structs or fields are modified.
 */
 
+// Represents a disk created on a cluster level based on PersistentVolumeClaim
+// --
+// +k8s:openapi-gen=true
+type HostDisk struct {
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace.
+	// Directly attached to the vmi via qemu.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#persistentvolumeclaims
+	// +optional
+	PersistentVolumeClaim *v1.PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
```

Comment:
I think we can model `HostDisk` pretty close to `HostPath` from `k8s: https://kubernetes.io/docs/concepts/storage/volumes/#hostpath

So something like:

```yaml
  - name: test-disk
    hostPath:
      # path to disk
      path: /data/disk.img
      # this says that the disk must be there
      type: Disk
```

and

```yaml
  - name: test-disk
    hostPath:
      # path to disk
      path: /data/disk.img
      # this says that the disk will be created if it is not there
      type: DiskOrCreate
      # disk size which will be used if the disk needs to be created
      capacity: 20Gi
```

Reply:
When we go that route, we would also have to consider the `HostDisk` if it exists on the VMI itself natively by adding a HostPath mount to the pod.

Reply:
Ok cool, and just to be on the same page, I understand from examples that you want to use hostDisk name in yaml definition (since hostPath type already exists), so it should look like this:
```
volumes:
  - name: test-disk
    hostDisk:
      path: /data/disk.img
      type: DiskOrCreate
      capacity: 20Gi
```
am I right?

Reply:
Yes, the snipped you provided would be translated to something like this on the pod template:

```
volumes:
  - name  test-disk
    hostPath:
      path: /data
      type: DirectoryOrCreate
```